app/assets/javascripts/discourse/app/static/markdown-it/features.js

```diff
@@ -0,0 +1,25 @@
+export default function loadPluginFeatures() {
+  const features = [];
+
+  for (let moduleName of Object.keys(requirejs.entries)) {
+    if (moduleName.startsWith("discourse/plugins/")) {
+      // all of the modules under discourse-markdown or markdown-it
+      // directories are considered additional markdown "features" which
+      // may define their own rules
+      if (
+        moduleName.includes("/discourse-markdown/") ||
+        moduleName.includes("/markdown-it/")
+      ) {
+        const module = requirejs(moduleName);
+
+        if (module && module.setup) {
+          const id = moduleName.split("/").reverse()[0];
+          const { setup, priority = 0 } = module;
+          features.push({ id, setup, priority });
+        }
+      }
+    }
+  }
+
+  return features;
+}
```

Comment:
This was previously inlined into `pretty-text/addon/engines/discourse-markdown-it.js`. I don't want the library/addon to reach into loader.js so this is factored out and pushed into the app.